power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/linear_pf_solver.hpp

```diff
@@ -87,29 +87,12 @@ template <symmetry_tag sym> class LinearPFSolver {
     // sparse linear equation
     ComplexTensorVector<sym> mat_data_;
     // sparse solver
-    SparseLUSolver<ComplexTensor<sym>, ComplexValue<sym>, ComplexValue<sym>> sparse_solver_;
-    typename SparseLUSolver<ComplexTensor<sym>, ComplexValue<sym>, ComplexValue<sym>>::BlockPermArray perm_;
+    using SparseSolverType = SparseLUSolver<ComplexTensor<sym>, ComplexValue<sym>, ComplexValue<sym>>;
+    SparseSolverType sparse_solver_;
+    typename SparseSolverType::BlockPermArray perm_;
```

Comment:
```suggestion
    using SparseSolverType = SparseLUSolver<ComplexTensor<sym>, ComplexValue<sym>, ComplexValue<sym>>;
    using BlockPermArray = typename SparseSolverType::BlockPermArray;

    SparseSolverType sparse_solver_;
    BlockPermArray perm_;
```

or move the `using` declarations to top of class